src/frontend/src/pages/GanttPage/GanttPageWrapper.tsx

```diff
@@ -7,71 +7,202 @@ import LoadingIndicator from '../../components/LoadingIndicator';
+        displayOrder: projects.find((p) => p.workPackages.find((w) => w.id === wp.id))!.id,
+        onClick: () => {
+          window.open(`/projects/${wbsPipe(wp.wbsNum)}`, '_blank');
+        }
+      };
+    };
+    if (projects) {
+      const filteredProjects = filterGanttProjects(projects, showCar1, showCar2, status, selectedTeam, start, end);
+      const projTasks = filteredProjects.map(transformProjectToGanttObject);
+      const workPackages = filteredProjects.flatMap((p) => p.workPackages);
+      const wpTasks = workPackages.map((wp) => transformWPToGanttObject(wp, filteredProjects));
+      setTeamList(Array.from(new Set(projects.map((p) => p.team?.teamName || 'No Team'))));
+      setGanttDisplayObjects([...projTasks, ...wpTasks]);
+    }
+  }, [end, expanded, projects, showCar1, showCar2, start, status, selectedTeam]);
 
-  if (projectIsError) return <ErrorPage message={projErr?.message} />;
+  if (isLoading) return <LoadingIndicator />;
 
-  if (wpIsError) return <ErrorPage message={wpErr?.message} />;
+  if (isError) return <ErrorPage message={error?.message} />;
 
-  const transformProjectToTask = (project: Project): Task => {
-    return {
-      id: wbsPipe(project.wbsNum),
-      name: wbsPipe(project.wbsNum) + ' ' + project.name,
-      start: project.startDate || new Date(),
-      end: project.endDate || new Date(),
-      progress:
-        (project.workPackages.filter((wp) => wp.status === WbsElementStatus.Complete).length / project.workPackages.length) *
-        100,
-      type: 'project',
-      hideChildren: true,
-      styles: { progressColor: '#e50000', backgroundColor: '#ff0000' },
-      displayOrder: project.id,
-      onClick: () => {
-        window.open(`/projects/${wbsPipe(project.wbsNum)}`, '_blank');
-      }
-    };
+  const car1Handler = (event: ChangeEvent<HTMLInputElement>) => {
+    const searchParams = `?status=${status}&showCar1=${event.target.checked}&showCar2=${showCar2}&selectedTeam=${selectedTeam}&expanded=${expanded}&start=${start}&end=${end}`;
+    history.push(`${history.location.pathname + searchParams}`);
   };
 
-  const transformWPToTask = (wp: WorkPackage, projects: Project[]): Task => {
-    return {
-      id: wbsPipe(wp.wbsNum), // Avoid conflict with project ids
-      name: wbsPipe(wp.wbsNum) + ' ' + wp.name,
-      start: wp.startDate,
-      end: wp.endDate,
-      progress: wp.progress,
-      project: projectWbsPipe(wp.wbsNum),
-      type: 'task',
-      styles: { progressColor: '#9c9c9c', backgroundColor: '#c4c4c4' },
-      displayOrder: projects.find((p) => p.workPackages.find((w) => w.id === wp.id))!.id,
-      onClick: () => {
-        window.open(`/projects/${wbsPipe(wp.wbsNum)}`, '_blank');
-      }
-    };
+  const car2Handler = (event: ChangeEvent<HTMLInputElement>) => {
+    const searchParams = `?status=${status}&showCar1=${showCar1}&showCar2=${event.target.checked}&selectedTeam=${selectedTeam}&expanded=${expanded}&start=${start}&end=${end}`;
```

Comment:
we build this search param everywhere with the same exact template which means it should be a function (like in the gantt.ts file!)